File: Daily Challenge/January 2025/Find the Prefix Common Array of Two Arrays.cpp
class Solution {
public:
  vector<int> findThePrefixCommonArray(vector<int>& A, vector<int>& B) {
    int n = A.size();
    vector<bool> markA(n + 1, false);
    vector<bool> markB(n + 1, false);
    int counter = 0;
    vector<int> ans(n, 0);

    for (int i = 0; i < n; ++i) {
      if (markA[B[i]]) counter++;
      markB[B[i]] = true;
      if (markB[A[i]]) counter++;
      markA[A[i]] = true;

      ans[i] = counter;
    }

    return ans;
  }
};

File: Daily Challenge/January 2025/Find the Prefix Common Array of Two Arrays.cs
public class Solution {
  public int[] FindThePrefixCommonArray(int[] A, int[] B) {
    int n = A.Length;
    bool[] markA = new bool[n + 1];
    bool[] markB = new bool[n + 1];
    int counter = 0;
    int[] result = new int[n];

    for (int i = 0; i < n; ++i) {
      if (markA[B[i]]) counter++;
      markB[B[i]] = true;
      if (markB[A[i]]) counter++;
      markA[A[i]] = true;

      result[i] = counter;
    }

    return result;
  }
}